### PR TITLE
EZP-25603: Implement a basic grid view

### PR DESCRIPTION
**Pull request by dpobel**
JIRA: https://jira.ez.no/browse/EZP-25603

# Description

~~(Note: depends on https://github.com/ezsystems/PlatformUIBundle/pull/533 which is not merged yet, only the last 4 commits need to be looked at but that's not very handy...)~~

Based on the subitem extensible view systems introduced in https://github.com/ezsystems/PlatformUIBundle/pull/533, this patch brings the grid view where locations/content items are displayed in a more visual way in boxes. For now, that's not very useful because all the grid items look the same since the grid item view is made as simple as possible and since we don't have yet a custom icon per content type. Those improvements will be implemented (shortly hopefully) in [EZP-25604](https://jira.ez.no/browse/EZP-25604) and [EZP-23022](https://jira.ez.no/browse/EZP-23022).

Screencast:

[](http://www.youtube.com/watch?v=LtyUIYE3ozU "Click to play on Youtube.com")

# Tasks

* [x] Getting https://github.com/ezsystems/PlatformUIBundle/pull/533 merged
* [x] Move/Rename subitem view template
* [x] Move/Rename subitem stylesheets
* [x] Loading error handling

### CODE REVIEW

**Comment by dpobel** [comment]
ping @StephaneDiot @yannickroger

File: Resources/public/css/theme/views/subitem/grid.css [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+
+.ez-view-subitemgridview .ez-subitemgrid-item-content {
+    /*background: #fff;*/
```
Comment (StephaneDiot) [inline Resources/public/css/theme/views/subitem/grid.css]: :)

File: Resources/public/css/theme/views/subitem/grid.css [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+.ez-view-subitemgridview .ez-subitemgrid-item-visual {
+    background: #fff;
+    border: 1px solid #333;
+	font-size: 4em;
+}
+
+.ez-view-subitemgridview .ez-subitemgrid-item-visual:before {
+	display: block;
```
Comment (StephaneDiot) [inline Resources/public/css/theme/views/subitem/grid.css]: indentation problems (same for the 2 on 3 rules after).

File: Resources/public/css/views/subitem/grid.css [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+.ez-view-subitemgridview .ez-subitemgrid-item-content-type {
+    margin: 0 0 0.3em 0;
+}
+
+.ez-view-subitemgridview .ez-subitemgrid-item-name {
+    margin: 0.3em 0 0 0;
+}
+
+.ez-view-subitemgridview .ez-subitemgrid-item-visual {
+	display: block;
```
Comment (StephaneDiot) [inline Resources/public/css/views/subitem/grid.css]: indentation.

File: Resources/public/js/views/subitem/ez-subitemgridview.js [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+        },
+
+        destructor: function () {
+            this._gridItemViews.forEach(function (item) {
+                item.destroy();
+            });
+        },
+
+        /**
+         * handles the loading error. It fires a notification and make sure the
```
Comment (StephaneDiot) [inline Resources/public/js/views/subitem/ez-subitemgridview.js]: Niptick^10: "**H**andles the loading ....."

File: Resources/public/templates/subitem/griditem.hbt [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<div class="ez-subitemgrid-item-content">
+    <h4 class="ez-subitemgrid-item-content-type">
+        <a href="{{path 'viewLocation' id=location.id languageCode=content.mainLanguageCode}}" class="ez-subitemgrid-item-link" data-icon="&#xe601;">
+            {{ contentType.names.[eng-GB] }}
+        </a>
+    </h4>
+    <a href="{{path 'viewLocation' id=location.id languageCode=content.mainLanguageCode}}" class="ez-subitemgrid-item-link ez-subitemgrid-item-visual" data-icon="&#xe601;"></a>
+    <h3 class="ez-subitemgrid-item-name">
+		<a href="{{path 'viewLocation' id=location.id languageCode=content.mainLanguageCode}}" class="ez-subitemgrid-item-link">
```
Comment (yannickroger) [inline Resources/public/templates/subitem/griditem.hbt]: Nitpick: indentation of the whole `<h3>` block

Comment (sunpietro) [inline Resources/public/templates/subitem/griditem.hbt]: Just my 2 cents, the whole structure of the template looks weird. How would you explain it semantically? What does it say?

**Comment by StephaneDiot** [comment]
+1 besides nipticks

File: Resources/public/css/views/subitem/grid.css [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+
+.ez-view-subitemgridview {
+    margin: 1em;
+}
+
+.ez-view-subitemgridview .ez-subitemgrid-content {
+    display: flex; /* TODO vendor prefix */
```
Comment (yannickroger) [inline Resources/public/css/views/subitem/grid.css]: Could we add those prefix now ? (same in block below)

File: Resources/public/css/views/subitem/grid.css (hunk 3 above)
Comment (yannickroger) [inline Resources/public/css/views/subitem/grid.css]: @StephaneDiot you mean "Indentation" :)
cf https://github.com/ezsystems/PlatformUIBundle/pull/541/files#r59334257

File: Resources/public/js/views/subitem/ez-subitemgridview.js (hunk 4 above)
Comment (yannickroger) [inline Resources/public/js/views/subitem/ez-subitemgridview.js]: Nitpick combo: makeS

**Comment by yannickroger** [comment]
+1 with feedback taken into account.

File: Resources/public/js/views/subitem/ez-subitemgridview.js [hunk 7]
```diff
@@ -0,0 +1,386 @@
...
+
+                this._gridItemViews.push(itemView);
+                gridContent.append(
+                    itemView.render().get('container')
+                );
+            }, this);
+        },
+
+        /**
+         * tap event handler on the load more button.
```
Comment (StephaneDiot) [inline Resources/public/js/views/subitem/ez-subitemgridview.js]: please backquote or capital letter (... https://github.com/ezsystems/PlatformUIBundle/pull/543/files#r59360070 :yum: )

File: Resources/public/templates/subitem/griditem.hbt (hunk 5 above)
Comment (dpobel) [inline Resources/public/templates/subitem/griditem.hbt]: very good remark, the `h4` should probably be a `p` or a `div` instead.

**Comment by dpobel** [comment]
Processed feedback, waiting for TravisCI to merge

**Pull request merged**